daemon/src/maker_cfd_actor.rs

```diff
@@ -292,24 +301,136 @@ impl MakerCfdActor {
         let cfd = Cfd::new(
             current_order.clone(),
             msg.quantity,
-            CfdState::Accepted {
+            CfdState::IncomingOrderRequest {
                 common: CfdStateCommon {
                     transition_timestamp: SystemTime::now(),
                 },
+                taker_id: msg.taker_id,
             },
         );
         insert_cfd(cfd, &mut conn).await?;
 
+        self.cfd_feed_actor_inbox
+            .send(load_all_cfds(&mut conn).await?)?;
+
+        Ok(())
+    }
+}
+
+#[async_trait]
+impl Handler<AcceptOrder> for MakerCfdActor {
+    async fn handle(&mut self, msg: AcceptOrder, _ctx: &mut Context<Self>) -> Result<()> {
+        tracing::debug!(%msg.order_id, "Maker accepts an order." );
+
+        let mut conn = self.db.acquire().await?;
+
+        // Validate if order is still valid
+        let cfd = match self.current_order_id {
+            Some(current_order_id) if current_order_id == msg.order_id => {
+                load_cfd_by_order_id(msg.order_id, &mut conn).await?
+            }
+            _ => {
+                // TODO: Return an error here?
```

Comment:
Our latest idea is that we have a dedicated feed where we can send error messages to the UI. We can add that later!

Reply:
For now, we can just return an error here using `bail!` if you want to abort.